content/ui-ux.md

```diff
@@ -0,0 +1,514 @@
+ 2. Fetch data from those subscriptions.
+ 3. Fetch global client-side state from stores such as the Router, Accounts, and your own stores.
+
+Ideally, once a smart component has assembled such a set of data, it passes it off to a pure component child to render with. The smart component actually does not render anything apart from one or more pure children.
+
+A typical use case for a smart component is the "page" component that the router points you to when you access a URL. Such a component typically needs to do the three things above and then can pass the resulting arguments into child components. In the Todos example app, the `listShowPage` does exactly this, with a resultingly simple template:
+
+```
+<template name="listsShowPage">
+  {{#each listIdArray}}
+    {{> listsShow todosReady=Template.subscriptionsReady list=list}}
+  {{/each}}
+</template>
+```
+
+## Visually testing pure components
+
+A useful property of pure components is that you can render them anywhere because they don't rely on complicated environments. One very useful thing that this enables is a component _styleguide_ or harness.
+
+A styleguide consists of two parts:
+
+1. A list of "entries"---a component coupled with a list of specifications; basically a list of different arguments that the component expects which trigger different behaviour.
+
+2. A special route in the development version of the application that renders one or more components with one or more of the specificiations.
+
+For instance, in Galaxy, we have a component styleguide that renders each pure component either one specification at a time, or with all specifications at once. 
+
+[ss]
+
+Such rendering enables very quick development of visual aspects of the component in all possible states. Typically in a complex application, it can be quite difficult to achieve certain states of components purely by "using" the application. For example, in Galaxy, the component screen enters quite a complex state if two deploys to the same application are happening simutaneously. This is a very hard to state to reach for a extended period of time if you are simply using the application!
+
+[ss]
+
+XXX: could say a lot more about this, but I feel like we might just blog about it / do a whole article on it's own once we've had a chance to release some code.
+
+## User Interface Patterns
+
+There are some patterns for building user interfaces that are useful to keep in mind for a Meteor application.
+
+### Responsive Design
+### Accessibility
+
+XXX: Honestly I'm really not sure there's very much Meteor specific in these two headings. I'm not convinced they belong in the guide
+
+
+### Internationalization
+
+Internationalization (i18n) is the process of generalizing the UI of your app in such a way that it's easy to render all text in a different language. In Meteor, the excellent `tap:i18n` package provides an API for building translations and using them in your templates and elsewhere.
```

Comment:
So I ran into an issue - tap:i18n actually depends on React, and doesn't work well with our current situation: https://github.com/TAPevents/tap-i18n/issues/115

Maybe not relevant to the current version of the guide, but perhaps a code item if we want most of the guide to be relevant for React and Blaze.